frontend/src/components/ExpenseItem.js

```diff
@@ -25,11 +25,19 @@ class ExpenseItem extends React.Component {
 
   render() {
     const { expenseItem } = this.props;
+
+    const getCategoryName = (id) => {
+      const categories = require("../phrases/Categories.json").find(
+        (el) => el.categoryId === Number(id)
+      );
+      return categories["categoryName"]; // returns the name of category from CategoryId
+    };
+
     return (
       <Container as={Card.Body} className="p-2 mt-2 transactionContainer ">
         <Row className="d-flex justify-content-between align-items-center">
           <Col xs={7} className="pr-0">
-            <p>{expenseItem.category}</p>
+            <p>{getCategoryName(expenseItem.category)}</p>
           </Col>
           <Col className="px-0 d-flex justify-content-end">
             <p>{expenseItem.amount} €</p>
```

Comment:
Technically this already does what it's supposed to, but just to give you an idea how to approach the refactoring, here's something that could be done.

Your current solution has a getCategoryName(id) function which gets *the* category object corresponding to the given id:
```js
const categories = require("../phrases/Categories.json").find(
  (el) => el.categoryId === Number(id)
);
```
So the `const categories` variable seems to be named a bit wrong, it's a single category:
```js
const category = require("../phrases/Categories.json").find(
  (el) => el.categoryId === Number(id)
);
```
Although further, the whole variable seems unnecessary, since it's only used once, in the return statement to get the category name:
```js
{
  const categories = require("../phrases/Categories.json").find(
    (el) => el.categoryId === Number(id)
  );
  return categories["categoryName"]; // returns the name of category from CategoryId
};
```
We could just return the category name directly:
```js
{
  return require("../phrases/Categories.json").find(
    (el) => el.categoryId === Number(id)
  )["categoryName"]; // returns the name of category from CategoryId
};
```
Also, I feel the `object.property` notation would be clearer than the `object[property]`. Both work since we know the property name. So:
```js
{
  return require("../phrases/Categories.json").find(
    (el) => el.categoryId === Number(id)
  ).categoryName; // returns the name of category from CategoryId
};
```
So now the whole getCategoryName arrow function would look like:
```js
const getCategoryName = (id) => {
  return require("../phrases/Categories.json").find(
    (el) => el.categoryId === Number(id)
  ).categoryName; // returns the name of category from CategoryId
};
```
Since the arrow function only has a return statement, it can be further cleaned by dropping the block wrapping and return keyword:
```js
const getCategoryName = (id) => require("../phrases/Categories.json").find(
  (el) => el.categoryId === Number(id)
).categoryName; // returns the name of category from CategoryId
```
Now getCategoryName is about as terse as it's going to get.  

However, if we look at it's use, there's only one call:
```js
const getCategoryName = (id) =>
  require("../phrases/Categories.json").find(
    (el) => el.categoryId === Number(id)
  ).categoryName; // returns the name of category from CategoryId

return (
  <Container as={Card.Body} className="p-2 mt-2 transactionContainer ">
    <Row className="d-flex justify-content-between align-items-center">
      <Col xs={7} className="pr-0">
        <p>{getCategoryName(expenseItem.category)}</p>
      </Col>
```
So putting it in a separate function is a bit unnecessary. We could just call the content code of the arrow function once where we need it. Then the whole render function would look like:
```js
render() {
  const { expenseItem } = this.props;

  return (
    <Container as={Card.Body} className="p-2 mt-2 transactionContainer ">
      <Row className="d-flex justify-content-between align-items-center">
        <Col xs={7} className="pr-0">
          <p>
            {
              require("../phrases/Categories.json").find(
                (el) => el.categoryId === Number(expenseItem.category)
              ).categoryName
            }
          </p>
        </Col>
```